src/fu-main.c

```diff
@@ -174,15 +174,49 @@ fu_main_engine_percentage_changed_cb (FuEngine *engine,
 				       g_variant_new_uint32 (percentage));
 }
 
+static gboolean
+fu_main_get_device_flags_for_sender (FuMainPrivate *priv, const char *sender,
+				     FwupdDeviceFlags *flags, GError **error)
+{
+	uid_t calling_uid;
+	g_autoptr(GVariant) value = NULL;
+
+	g_return_val_if_fail (sender != NULL, FALSE);
+	g_return_val_if_fail (flags != NULL, FALSE);
+
+	value = g_dbus_proxy_call_sync (priv->proxy_uid,
+					"GetConnectionUnixUser",
+					g_variant_new ("(s)", sender),
+					G_DBUS_CALL_FLAGS_NONE,
+					2000,
+					NULL,
+					error);
+	if (value == NULL)
+		return FALSE;
```

Comment:
i think you probably want a g_prefix_error here to give some context. Either here or in fu_main_device_array_to_variant.